tests/unity/unity_test.cmake

```diff
@@ -0,0 +1,135 @@
+#
```

Comment:
Is it possible to avoid this file by organizing the CMock generation as a Zephyr library / subsystem ?

I have not seen anything so far that couldn't be moved out of this file at least ...

Reply:
i don't know. im a cmake newbie. but wouldn't that require to move that upstream? I'm not sure if we want to keep it ncs or move to zephyr.

Reply:
No, it wouldn't require moving to upstream. nrf can define Zephyr libraries.

Generally, for new software, we want to organize it as libraries. And then have application's enable that software by including boilerplate.cmake, and then enabling it through Kconfig.

This PR uses a different boilerplate.cmake to express that a piece of SW should be enabled. This is unfortunate because it diverges from how we generally organize things.

Now, it could be that we for technical reasons need to do so, but before we do something different, we need to try to stick to the system.

Reply:
I moved everything to CMAkeLists.txt and removed unity_test.cmake. I had one issue: when i'm setting variable in library it is not seen in functions. For time being i've added copy of those variables in functions. How to define variables that will have global scope?

Reply:
Discussed offline